src/api/python/wheels/build_wheel.py

```diff
@@ -0,0 +1,155 @@
+import platform
+import subprocess
+import multiprocessing
+import shutil
+import glob
+
+from setuptools import setup, Extension
+from setuptools.command.build_ext import build_ext
+from distutils.version import LooseVersion
+
+def get_project_src_path():
+    # expecting this script to be in src/api/python/wheels
+    # The project source directory is 4 directories up
+    # (5 dirnames because start with file)
+    project_src_path = __file__
+    for i in range(5):
+        project_src_path = os.path.dirname(project_src_path)
+    return os.path.abspath(project_src_path)
+
+
+def get_cvc5_version():
+    project_src_path = get_project_src_path()
+
+    # read CMakeLists.txt to get version number
+    version = dict()
+    str_pattern = 'set\(CVC5_(?P<component>MAJOR|MINOR|RELEASE)\s*(?P<version>\d+)\)'
+    pattern = re.compile(str_pattern)
+    with open(project_src_path + '/CMakeLists.txt', 'r') as f:
+        for line in f:
+            line = line.strip()
+            m = pattern.search(line)
+            if m:
+                gd = m.groupdict()
+                version[gd['component']] = gd['version']
+                if len(version) == 3:
+                    break
+
+    assert len(version) == 3, 'Expecting MAJOR, MINOR and RELEASE'
+    return version['MAJOR'], version['MINOR'], version['RELEASE']
+
+
+class CMakeExtension(Extension):
+    def __init__(self, name, sourcedir=''):
+        Extension.__init__(self, name, sources=[])
+        self.sourcedir = os.path.abspath(sourcedir)
+
+
+class CMakeBuild(build_ext):
+    def run(self):
+        try:
+            out = subprocess.check_output(['cmake', '--version'])
+        except OSError:
+            raise RuntimeError(
+                "CMake must be installed to build the following extensions: " +
+                ", ".join(e.name for e in self.extensions))
+
+        if self.is_windows():
+            cmake_version = LooseVersion(
+                re.search(r'version\s*([\d.]+)', out.decode()).group(1))
+            if cmake_version < '3.1.0':
+                raise RuntimeError("CMake >= 3.1.0 is required on Windows")
+
+        for ext in self.extensions:
+            self.build_extension(ext)
+
+    @staticmethod
+    def is_windows():
+        tag = platform.system().lower()
+        return tag == "windows"
+
+    @staticmethod
+    def is_linux():
+        tag = platform.system().lower()
+        return tag == "linux"
+
+    def build_extension(self, ext):
+        extdir = os.path.abspath(
+            os.path.dirname(self.get_ext_fullpath(ext.name)))
+
+        cfg = 'Production'
+        build_args = ['--config', cfg]
+
+        cpu_count = max(2, multiprocessing.cpu_count() // 2)
+        build_args += ['--', '-j{0}'.format(cpu_count)]
+
+        project_src_path = get_project_src_path()
+        build_dir = os.path.join(project_src_path, "build")
+
+        # to avoid multiple build, only call reconfigure if we couldn't find the makefile
+        # for python
```

Comment:
Is this actually an issue? How do you make sure that the Makefile is up-to-date?

Reply:
Yeah it's not that big of a deal, I only kept it in because it was in the one from smt-switch tbh. It is possible that the Makefile is stale and configured with python. Maybe a better alternative would be to use a special build directory name? Something like `python-wheel`. And if it already exists, we could just throw an exception? I'm not sure it'd be a good idea to automatically delete a whole build folder even if it's a dedicated name.

Reply:
Why not just reconfigure and rebuild?

Reply:
We could do that. I just worried about messing up someone's existing build.

Reply:
How about creating a `build-python-wheel` directory and use this as working directory?

Reply:
Yeah sounds good. Updated with d4b75d15820298812c8acce8678fd614a726f7b9.